notifier/notify_windows.go

```diff
@@ -1,16 +1,290 @@
+			return
+			//TODO: what to do?
+		}
+	}
+}
+
+func initLogo() {
+	if _, err := os.Stat(logoLocation); err != nil { //File doesn't exist or another Error that is handled while copying file there
+		os.MkdirAll(filepath.Dir(logoLocation), os.ModePerm)
+		if err = downloadFile(logoLocation, webLogoLocation); err != nil {
+			log.Errorf("Can't copy Logo for SnoreToast from %s to %s: %s", webLogoLocation, logoLocation, err)
+		}
+	}
+}
+
+func notificationListener() {
+	log.Debugf("Listening on Pipe %s\n", pipeName)
+	l, err := winio.ListenPipe(pipeName, nil)
+	if err != nil {
+		log.Errorf("Error while starting Pipe for SnoreToast: %s", err)
+		return
+	}
+	defer l.Close()
+
+	for {
+		conn, err := l.Accept()
+		if err != nil {
+			log.Errorf(err)
+			continue
+		}
+		go onCallback(conn)
+	}
+}
+
+func onCallback(conn net.Conn) {
+	data := snoreToastStrToMap(readWstrConnToStr(conn)) //TODO: what  if some required attribute isn't there?
+	log.Debugf("onCallback: %+v", data)
+
+	if data["action"] == "timedout" {
+		return
+	}
+
+	notificationsLock.Lock()
+	var n *Notification
+	for _, elem := range notifications {
+		if elem.GUID == data["notificationId"] {
+			n = elem
+		}
+	}
+	notificationsLock.Unlock()
+	if n == nil {
+		return
+	}
+	n.Lock()
+	defer n.Unlock()
+
+	switch data["action"] {
+	case "dismissed": //do nothing
+	case "clicked": //like dismissed: do nothing
+	case "buttonClicked":
+		for _, actions := range n.AvailableActions {
+			if actions.Text == data["button"] {
+				n.SelectAction(actions.ID)
+			}
+		}
+	default:
+		log.Debugf("SnoreToast-Action %s not implemented yet", data["action"])
+	}
+}
+
+//Snoretoast helper-Functions:
+
+func snoreToastStrToMap(in string) map[string]string {
+	ret := make(map[string]string)
+
+	for _, el := range strings.Split(in, ";") { //TODO: make sure, there is no ; in any value!
+		if el == "" {
+			continue
+		}
+
+		kv := strings.Split(el, "=") //TODO: make sure, there is no = in any value!
+		if len(kv) != 2 {
+			log.Warning("SnoreToast-Callback: " + el + " cannot be parsed in Key=Value")
+			continue
+		}
+
+		ret[kv[0]] = kv[1]
+	}
+
+	return ret
+}
+
+func buildSnoreToastButtonArgument(a []*Action) string {
+	temp := make([]string, 0, len(a))
+	for _, elem := range a {
+		if elem.Text != "" {
+			temp = append(temp, elem.Text)
+		}
+	}
+	return strings.Join(temp, ";")
+}
+
+func snoreToastRunCmdAsync(cmd *exec.Cmd) {
+	go func() {
+		exit, err := execCmd(cmd)
+
+		switch exit {
+		case 0: //do nothing
+		case 1: //do nothing
+		case 2: //do nothing
+		case 3: //do nothing
+		case 4: //do nothing
+		case 5: //do nothing
+		default:
+			log.Errorf("Error while running %+v: %s", cmd.Args, err)
+		}
+	}()
+}
+
+//Generel helper-Functions:
+
+func execCmd(cmd *exec.Cmd) (exitCode int, err error) {
+	log.Debugf("Run snoretoast:%+v", cmd.Args)
+
+	err = cmd.Run()
+
+	//Used: https://stackoverflow.com/questions/10385551/get-exit-code-go
```

Comment:
Please do not use code directly from Stackoverflow. The license requirements (`cc by-sa 3.0 with attribution required`) aren't that simple and there might be certain incompatibilities with how we handle licensing at the moment. Sorry if I told you otherwise before.
Rather, please consult the official golang documentation and examples.